foodnet/membership/views.py

```diff
@@ -1,15 +1,181 @@
 # -*- coding: utf-8 -*-
-from django.shortcuts import render
-
 import logging
 
+from django.db import IntegrityError
+from django.http import Http404
+from django.shortcuts import render, redirect, get_object_or_404
+from django.core.urlresolvers import reverse, reverse_lazy
+from django.contrib import messages
+from django.contrib.auth import login, authenticate
+from django.contrib.auth.decorators import login_required, permission_required
+from django.views.generic import FormView
+
+from allauth.account.models import EmailConfirmation, EmailAddress
+from allauth.account.views import sensitive_post_parameters_m,\
+    PasswordSetView, PasswordChangeView
+
+from foodnet.common.views import LoginRequiredMixinView
+from .forms import ProfileForm, InviteForm, AcceptInvitationForm
+from .models import Invitation, User, UserProfile
+from .utils import create_verified_user
+
+
+log = logging.getLogger(__name__)
+
+
+@permission_required('membership.can_invite')
+@login_required
+def invite(request):
+    """Invite a new email address."""
+    form = InviteForm()
+    if request.method == 'POST':
+        form = InviteForm(request.POST)
+        if form.is_valid():
+            email = form.cleaned_data['email']
+            existing_user = User.objects.filter(email=email)
+            existing_email = EmailAddress.objects.get_users_for(email=email)
+            if existing_user or existing_email:
+                msg = 'User {} already exists.'.format(email)
+                messages.add_message(request, messages.ERROR, msg)
+            else:
+                Invitation.objects.create(
+                    email=email,
+                    invited_by=request.user,
+                    division=form.cleaned_data['division'],
+                    member_category=form.cleaned_data['member_category'])
+                msg = 'Invitation has been send to {}'.format(email)
+                messages.add_message(request, messages.SUCCESS, msg)
+                return redirect(reverse('home'))
+    ctx = {
+        'form': form,
+        'title': "send invitation",
+    }
+    return render(request, 'membership/invite.html', ctx)
 
-log = logging.getLogger(__file__)
 
+def accept_invitation(request, verification_key):
```

Comment:
The only I can think of is it could potentially be a few lines of code more and was worry it's going to be less easy to debug at first. It may be a good and easy redactor task for hack day, if we find some one who want's to get to know foodnet invitations and allauth etc(?)